test_room_allocation.py

```diff
@@ -96,11 +80,127 @@ def test_doesnt_add_staff_to_livingspace(self):
+        self.amity.create_room('Narnia~o', 'Hogwarts~l', 'Camelot~o')
+        # add 7 people
+        for person in people:
+            self.amity.add_person(person, 'staff')
+            self.amity.reallocate_person(person, 'Camelot')
+
+        camelot = self.amity.get_room_by_name('Camelot')
+        self.assertLess(len(camelot.people_in_room), 7)
+
+    def test_cannot_add_more_than_4_people_to_livingspace(self):
+        people = ['Paul', 'Ryan', 'Barney', 'Mark', 'Angie']
+        self.amity.create_room('Narnia~l', 'Hogwarts~l', 'Camelot~o')
+        # add 7 people
+        for person in people:
+            self.amity.add_person(person, 'fellow', wants_accommodation=True)
+            self.amity.reallocate_person(person, 'Hogwarts')
+        hogwarts = self.amity.get_room_by_name('Hogwarts')
+        self.assertLess(len(hogwarts.people_in_room), 5)
+
+    def test_cannot_reallocate_to_wrong_room(self):
+        self.amity.create_room("Camelot~o", "Hogwarts~l")
+        self.amity.add_person("Lancelot", 'staff')
+
+        status = self.amity.reallocate_person("Lancelot", 'Hogwarts')
+        self.assertTrue("cannot be in" in status)
+
+        self.amity.add_person("Arthur", 'fellow')  # doesn't want accommodation
+        status2 = self.amity.reallocate_person("Arthur", 'Hogwarts')
+        self.assertTrue("cannot be in" in status2)
+
+    def test_cannot_reallocate_non_existent_person(self):
+        self.amity.create_room('Hogwarts~l')
+        status = self.amity.reallocate_person('Dumbledore', 'Hogwarts')
+        self.assertEqual(status, "Person does not exist.")
+
+    def test_cannot_reallocate_to_non_existent_room(self):
+        self.amity.create_room("Camelot~o", "Hogwarts~l")
+        self.amity.add_person("Potter", "fellow")
+        status = self.amity.reallocate_person("Potter", "Non_Existent_Room")
+        self.assertEqual(status, "Room does not exist.")
+
+    def test_reallocate_removes_from_unallocated(self):
         pass
 
-    def test_cannot_add_more_than_4_to_livingspace(self):
+    def test_print_room_nonexistent_room_name(self):
+        status = self.amity.print_room("NonExistentRoom")
+        self.assertEqual(status, "Room does not exist.")
+
+    def test_print_allocations_no_rooms_exist(self):
+        status = self.amity.print_allocations()
+        self.assertEqual(status, "No rooms exist.")
+
+    def test_print_allocations_pipes_to_file(self):
+        person = 'Paul'
+        self.amity.create_room('Hogwarts~l')
+        self.amity.add_person(person, 'fellow', wants_accommodation=True)
+        self.amity.print_allocations(filename="test_amity")
+        repr_in_file = 'Hogwarts' + '\n' + ('-' * 37) +\
+            '\n' + 'Paul' + '\n'
+
+        directory = "test_files/"
+        f = open(directory + 'test_amity.txt', 'r')
+        data = f.read()
+        f.close()
+
+        self.assertEqual(data, repr_in_file)
+
+    def test_print_allocations_illegal_filename_chars(self):
```

Comment:
Test name not very descriptive. Maybe change illegal to something that is easy to understand.